Samples/WindowsCopilotRuntime/cs-wpf/WCRforWPF/MainWindow.xaml.cs

```diff
@@ -0,0 +1,177 @@
+﻿using System.Text;
+using System.Threading.Tasks;
+using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Data;
+using System.Windows.Documents;
+using System.Windows.Input;
+using System.Windows.Media;
+using System.Windows.Media.Imaging;
+using System.Windows.Navigation;
+using System.Windows.Shapes;
+using System.Drawing.Imaging;
+using Microsoft.Graphics.Imaging;
+using Microsoft.Windows.AI.Generative;
+using Microsoft.Windows.Vision;
+using Windows.Graphics.Imaging;
+using Windows.Storage;
+
+namespace WCRforWPF;
+
+/// <summary>
+/// Interaction logic for MainWindow.xaml
+/// </summary>
+public partial class MainWindow : Window
+{
+    public MainWindow()
+    {
+        InitializeComponent();
+    }
+
+    private async void SelectFile_Click(object sender, RoutedEventArgs e)
+    {
+        _currentImage = null;
+        this.Description.Text = "(picking a file)";
+        this.FileContent.Text = "(picking a file)";
+
+        var fileDialog = new Microsoft.Win32.OpenFileDialog();
+        fileDialog.DefaultExt = ".jpg";
+        fileDialog.Filter = "Image files (*.jpg, *.jpeg, *.png)|*.jpg;*.jpeg;*.png";
+        if (fileDialog.ShowDialog() == true)
+        {
+            this.FilePath.Text = fileDialog.FileName;
+            this.ProcessFile.IsEnabled = true;
+            _currentImage = await GetImageBufferFromFile(fileDialog.FileName);
+            var source = new BitmapImage(new Uri(fileDialog.FileName));
+            this.InputImage.Source = source;
+        }
+        this.Description.Text = "(done picking)";
+        this.FileContent.Text = "(done picking)";
+    }
+
+    private void PasteImageContent(object sender, RoutedEventArgs e)
+    {
+        this.Description.Text = "(pasting)";
+        this.FileContent.Text = "(pasting)";
+        var clipboard = System.Windows.Clipboard.GetDataObject();
+        if (clipboard == null)
+        {
+            this.FileContent.Text = "Can't open the clipboard, try again";
+            return;
+        }
+
+        var bitmapContent = clipboard.GetData(DataFormats.Bitmap, true) as BitmapImage;
+        if (bitmapContent == null)
+        {
+            this.FileContent.Text = "That wasn't a bitmap, try again";
+            return;
+        }
+    }
+
+    private ImageDescriptionGenerator? _ig;
+    private ImageBuffer? _currentImage;
+
+    private async Task<ImageBuffer> GetImageBufferFromFile(string path)
+    {
+        var storageFile = await StorageFile.GetFileFromPathAsync(path);
```

Comment:
Mental note with @jeffstall - we should get some "rosetta stone" set of "if you have a framework image of type X here is how you convert it to an imagebuffer" samples added.

For instance, StorageFile is not a "standard" WPF type - https://learn.microsoft.com/dotnet/desktop/wpf/graphics-multimedia/imaging-overview?view=netframeworkdesktop-4.8has a list of things like "decode a file to a png" and a way to get the raw pixels out.